src/storage/SingleThreadedResourceLocker.ts

```diff
@@ -1,13 +1,17 @@
+import { EventEmitter } from 'events';
 import AsyncLock from 'async-lock';
 import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
 import { getLoggerFor } from '../logging/LogUtil';
-import type { Lock } from './Lock';
-import type { ResourceLocker } from './ResourceLocker';
+import type { ExpiringLock } from './ExpiringLock';
+import type { ExpiringResourceLocker } from './ExpiringResourceLocker';
+
+/** Time in ms after which reading a representation times out, causing the lock to be released. */
+const READ_TIMEOUT = 1000;
 
 /**
  * A resource locker making use of the `async-lock` library.
  */
-export class SingleThreadedResourceLocker implements ResourceLocker {
+export class SingleThreadedResourceLocker implements ExpiringResourceLocker {
```

Comment:
There's no need to put the expiry functionality in `SingleThreadedResourceLocker`; I wouldn't change that class.

Rather, I would write an `ExpiringResourceLocker` whose constructor takes a `ResourceLocker` as an argument. Upon `aquire`, it gets a lock from the passed locker, and adds expiry functionality to it. In other words, the logic from below, but moved into another class such that _any_ lock (not just the current single-threaded lock) can become an expiring lock.

And READ_TIMEOUT can also be a constructor parameter then.